src/components/studentlist.tsx

```diff
@@ -0,0 +1,242 @@
+//import React from 'react';
+import React,{ useEffect } from 'react';
+import ListGroup from 'react-bootstrap/ListGroup';
+import LectureItem from "./lectureItem"
+import CourseItem from "./CourseItem"
+import StudentItem from "./studentItem"
+import axios from "axios";
+import Calendarweek from "./Calendarweek"
+
+
+
+
+  class studentList extends React.Component {
+  constructor(props) {
+    super(props);
+    
+    this.state={lectures:[],students:[],courses:[],selectedLecture:0,selectedCourse:0};
+ 
+  }
+  
+  getStudentList = (lectureID) => {
+     
+    axios  
+      .get(`http://localhost:4001/api/lectures/:${lectureID}/students`, {})
+      .then((response) => {
+        let result = response.data.reduce((acc, er) => er.name, 0);
+        this.setState({ lectures: result });
+
+        
+      })
+    
+  };
+
+  getCourseList = () => {
+     
+    axios  
+      .get(`http://localhost:4001/api/courses`, {})
+      .then((response) => {
+        let result = response.data.reduce((acc, er) => er.name, 0);
+        this.setState({ courses: result });
+
+        
+      })
+    
+  };
+
+  getLecturesList = (lectureID) => {
+     
+    axios  
+      .get(`http://localhost:4001/api/lectures/`, {
+
+        lectureid:lectureID,
+      })
+      .then((response) => {
+        let result = response.data.reduce((acc, er) => er.name, 0);
+        this.setState({ courses: result });
+
+        
+      })
+    
+  };
+
+  formatEvents() {
+    return this.state.lectures.map(lecture => {
+              const {name, end, start} = lecture
+  
+              let startTime = new Date(start)
+              let endTime = new Date(end)
+  
+              return {
+                title:name, 
+                start: startTime,
+                end: endTime, 
+                extendedProps: {...lecture}
+              }
+          })
+  }
+  handleEventClick= ({event}) => {
+    console.log("studentlistview");
+    this.scrolltoview("studentlistview")  ;
+}
+
+handleEventDrop = (info) => {
+        if(window.confirm("Are you sure you want to change the event date?")){
+            console.log('change confirmed')
+
+            // updateAppointment is another custom method
+            this.props.updateAppointment({...info.event.extendedProps, start: info.event.start, end: info.event.end})
+
+        } else {
+            console.log('change aborted')
+        }
+   }
+
+  scrolltoview=(elementId)=>{
+    var input=((typeof elementId)=="string")?  elementId:"calendarWeekContainer";
+
+    console.log(input);
+    document.getElementById(input).scrollIntoView({
+      behavior: 'smooth'
+  });
+  } 
+
+  componentDidMount(){
+    this.setState({selectedLecture:1});
+    this.getLecturesList(1);
+     
+   
+    this.setState({lectures:[ 
+      
+      {id:2323,
```

Comment:
Is this for testing? Can't we put this data in the DB?

Reply:
yes, We can, I just created my own data to check the GUI, but I decided not to change db because of merge problems , but we can put them in db in another commit.